Dbertozzi/feature shop detail edit #16

dbertozzi opened the pull request:
Created an alternate context for 'contact owner' button when shop owner is logged in. Clicking this button opens a dialog where owner may edit the shop description and title. Created a put endpoint for the update and added a request to front end. Cleaned up file structure to increase readability. Looking forward to getting feedback. I did not add anything to let the user know there is an error if put request is not successful and will appreciate to advice on where to add this.

dbertozzi committed:
Added Edit Shop Details Dialog

Found the check to see if the shop is the users not working. Fixed this issue. Modified 'Contact Owner' button to allow editing shop details when logged in. Dialog now appears to edit shop title and description. Requires an end point for the title and description still. I think I can change the file structure of this code somewhat. I would appreciate advice on this topic.

dbertozzi committed:
Added shop details put endpoint, started adding put request in front end

End point works with postman. Getting errors accessing from the client. Need to sort these out tomorrow.

dbertozzi committed:
Completed put request from front end and restructured files

Completed the put request for updating shop details. Modified the reducer to manage entire state. Moved components into individual files to improve readability.

naapak commented on client/src/components/shop/ProductCard.js:
```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useReducer, useState } from "react";
```
where are you using reducer and state here?

dbertozzi replied:
Good point. Must have been left there by accident when I was splitting up code.

naapak reviewed (approved):
Looks good to me.

dbertozzi committed:
Changes based on Naresh's Feedback

Removed unused dependencies

dbertozzi committed:
Added Endpoint for cover photo upload

Endpoint uploads new cover photo to S3, updates the shop cover photo reference in the database, and then deletes the old cover photo from S3.
Additionally I cleaned up a few small things.

skassam21 commented on client/src/components/shop/EditCoverDialog.js:
```diff
@@ -0,0 +1,46 @@
+import DialogTitle from "@material-ui/core/DialogTitle";
+import { useDropzone } from "react-dropzone";
+
+function CoverDropzone() {
```
I like... maybe should be in another file for reusable components

skassam21 commented on client/src/components/shop/ShopBanner.js:
```diff
@@ -0,0 +1,81 @@
+  const classes = useStyles();
+  return (
+    <Grid container component="main">
+      <CssBaseline />
```
Why is CssBaseline here?

dbertozzi replied:
I think because I copied code from a template. Will look into it.

skassam21 commented on client/src/pages/Shop.js:
```diff
@@ -151,115 +51,94 @@ export default function Shop() {
+
     fetchShop();
   }, []);
+  const UPDATE_DETAILS_API = "http://localhost:3001/shop/details-update/";
```
Don't use localhost:3001, use a relative path like `/shop/details-update/`

skassam21 commented on client/src/pages/Shop.js:
```diff
@@ -151,115 +51,94 @@ export default function Shop() {
+  const UPDATE_DETAILS_API = "http://localhost:3001/shop/details-update/";
+  async function updateDetails() {
+    const config = {
+      headers: { "auth-token": token }
```
did we ever create a utils to do these auth requests?

dbertozzi replied:
Naresh created a request util but it was not yet merged when I began.

skassam21 commented on client/src/components/shop/ProductCard.js:
```diff
@@ -0,0 +1,96 @@
+                : { display: "initial" }
+            }
+            onClick={() =>
+              props.dispatch({ type: "PRODUCT_LIKED", likeIndex: props.index })
```
In the future this will make API calls?

dbertozzi replied:
I think we will have to store what a user has liked in their user model. We would likely have a page where they can view their liked content. That is the point of likes or wishlist i think.

skassam21 commented on client/src/pages/Shop.js:
```diff
@@ -134,11 +30,15 @@ export default function Shop() {
   // fetch shop data
   const token = localStorage.getItem("token");
+
   const USER_API = "http://localhost:3001/users/";
```
I know this is not your code, but also here not to use http://localhost:3001

dbertozzi replied:
This was my code. I have removed localhost from all API paths in shop.

skassam21 commented on server/routes/shop.js:
```diff
@@ -141,7 +129,31 @@ router.post(
   }
 );
-
+router.put("/details-update/:userid", verify, async (req, res) => {
```
I feel like there is a better naming convention that is more RESTful for this, something like:
PUT `shops/:shop_id/details` or userid instead of shopid

skassam21 commented on server/routes/shop.js:
```diff
@@ -184,4 +196,50 @@ router.post("/item/delete-image/:itemid", verify, async (req, res, next) => {
     });
 });
 
+router.put("/cover-update/:userid", verify, singleUpload, async (req, res) => {
```
same as above, rather it be: `shops/:shop_id/cover`

skassam21 replied:
You don't need upate in the URL, since a put already means update

dbertozzi replied:
if I move the path variable to the middle I have to use more variables in the string template. `${API_ROOT}${userid}${API_EXTENSION}` instead of `${API}${userid}`

skassam21 commented on server/routes/shop.js:
```diff
@@ -184,4 +196,50 @@ router.post("/item/delete-image/:itemid", verify, async (req, res, next) => {
+  });
+  console.log(coverUpdate);
+  const Key = oldCoverPhoto.split("/").slice(-1)[0];
+  const imageDeleted = deleteImage({ Key: Key });
```
For now this is okay, ideally we keep them in the future

dbertozzi committed:
Added request from Client to update cover

Implimented changes suggested by Shums' review and added put request using axios from client. Requested changes included removing localhost from api routes, moving dropzone to a seperate file, and modifying API URLs.

dbertozzi committed:
Merge branch 'dev' into dbertozzi/feature-shop-detail-edit-#16
